Towny/src/main/java/com/palmergames/bukkit/towny/object/Resident.java

```diff
@@ -593,9 +593,26 @@ public boolean hasTownRank(String rank) {
 	}
 
 	public List<String> getTownRanks() {
+		if (TownyPerms.ranksWithTownLevelRequirementPresent() && !townRanks.isEmpty() && hasTown()) {
+			ArrayList<String> out = new ArrayList<>();
+			for (String rank : new ArrayList<>(townRanks)) {
+				int requiredTownLevelForRank = TownyPerms.getRankTownLevelReq(rank);
+				if (requiredTownLevelForRank == 0 || getTownOrNull().getLevelNumber() >= requiredTownLevelForRank)
```

Comment:
Cache level number here and in the nation rank method in a variable that's outside the rank loop